pkg/tagger/collectors/ecs_main.go

```diff
@@ -9,24 +9,32 @@ package collectors
 
 import (
 	"fmt"
+	taggerutil "github.com/DataDog/datadog-agent/pkg/tagger/utils"
 	ecsutil "github.com/DataDog/datadog-agent/pkg/util/ecs"
+	"time"
 )
 
 const (
 	ecsCollectorName = "ecs"
+	ecsExpireFreq    = 5 * time.Minute
 )
 
 // ECSCollector listen to the ECS agent to get ECS metadata.
 // And feed a stream of TagInfo.
-
 type ECSCollector struct {
-	infoOut chan<- []*TagInfo
+	infoOut    chan<- []*TagInfo
+	expire     *taggerutil.Expire
+	lastExpire time.Time
+	expireFreq time.Duration
 }
 
 // Detect tries to connect to the ecs agent
 func (c *ECSCollector) Detect(out chan<- []*TagInfo) (CollectionMode, error) {
 	if ecsutil.IsInstance() {
 		c.infoOut = out
+		c.lastExpire = time.Now()
+		c.expireFreq = ecsExpireFreq
+		c.expire, _ = taggerutil.NewExpire(ecsExpireFreq)
```

Comment:
either handle the error or remove it from the constructor signature

Reply:
👌